### PR TITLE
Bella shim

### PR DESCRIPTION
**Pull request by kyshim**
@Howon  please read and roast
Lines 219-236

### CODE REVIEW

**Comment by Howon** [comment]
@kyshim  pull from Nick's branch please

File: script/db/index.js [hunk 1]
```diff
@@ -213,6 +213,27 @@ module.exports = firebase => {
...
+		readPlayListData: (uid) => {
+			const userRef = firebaseRef.child("users/" + uid);
+			const userPlayLists = userRef.child("playListIDs");
+			//make new map to store data structure of all playlists
+			const playListMap = new Map();
+			console.log(userPlayLists);
+
+			Object.entries(userPlayLists).forEach((_, value) => {
+				console.log(value);
+				let  returnValue = playListRef.child(value).once("value").then(snapshot => {
```
Comment (alanhdu) [inline script/db/index.js]: Is this a future I see? (or are they called promises in javascript land? Naming things is hard...). I never thought I'd see the day when javascript went beyond callbacks (now we just need to wait for `async` and `await`...)

File: script/db/index.js [hunk 2]
```diff
@@ -1,239 +1,239 @@
...
-						});
-					} else { //Case 3: Default user without any playList IDs has ["filter"] as a value
-						const newList = []; //Holds the final list being updated
-						const playListKey = playListRef.push(playListInput).key;
-
-						newList.push(playListKey);
-
-						userRef.update({
-							playListIDs: newList
+		const firebaseRef = firebase.database().ref(); //Reference to the entire db
```
Comment (alanhdu) [inline script/db/index.js]: Can we separate out reformatting from semantic changes? Mixing them in a single commit makes it much harder to review than it needs to be. Break it up into multiple commits.

(also, why are we using two tabs here in the first place? I thought the javascript standard was two-space indent?).

(Also, pro-tip: you can add `?w=1` at the end of the github URL to git a diff that ignores pure-whitespace changes).

Comment (Howon) [inline script/db/index.js]: it should be 2 space tab but I think they didnt change their sublime settings to do that

Comment (Howon) [inline script/db/index.js]: Please do this at bottom right corner
 
screen shot 2017-02-18 at 1 03 11 pm

File: script/db/index.js (hunk 1 above)
Comment (Howon) [inline script/db/index.js]: ES6 is nice

**Comment by Howon** [comment]
yall need to pull your branches together asap 

@kyshim @liunick @francesar @chanpham97

**Pull request merged**